### PR TITLE
Add Party

### PR DESCRIPTION
**Pull request by GoekhanAgkurt**

### CODE REVIEW

**Review by toshydev — APPROVED** [review]
Looks good to me! 🎉

**Comment by Pia0912** [comment]
this looks awesome! 🚀

**Comment by CananOzyol** [comment]
looks so good 🥹

File: backend/src/main/java/org/partypets/backend/service/PartyService.java [hunk 1]
```diff
@@ -14,7 +15,16 @@ public class PartyService {
 
     private final PartyRepo partyRepo;
 
+    private final UuIdService uuIdService;
```
Comment (Satttoshi) [inline backend/src/main/java/org/partypets/backend/service/PartyService.java:18]: UUID and its createUUID methods are a util class of java itself. It is redundant to have a Util class, utilizing another Util class for doing the same. It is legit though if you intend to do some advanced UUID handling in future commits.

File: frontend/src/components/AddForm.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import Button from "@mui/material/Button";
+
+
+type Props = {
+    onAddParty: (data: { date: string; location: string; theme: string }) => void;
+}
+
+export default function AddForm(props: Props) {
+
+    const [theme, setTheme] = useState("");
```
Comment (Satttoshi) [inline frontend/src/components/AddForm.tsx:12]: You may want to use generic typing, to utilize TypeScript for useState hook

```suggestion
    const [theme, setTheme] = useState<string>("");
```

File: frontend/src/components/AddForm.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    const [date, setDate] = useState("");
```
Comment (Satttoshi) [inline frontend/src/components/AddForm.tsx:13]:
```suggestion
    const [date, setDate] = useState<string>("");
```

File: frontend/src/components/AddForm.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    const [location, setLocation] = useState("");
```
Comment (Satttoshi) [inline frontend/src/components/AddForm.tsx:14]:
```suggestion
    const [location, setLocation] = useState<string>("");
```

File: frontend/src/components/PartyItem.tsx [hunk 5]
```diff
@@ -1,19 +1,32 @@
...
+                <CardContent>
+                    <Typography variant="overline" component="div">
+                        {props.party.theme}
+                    </Typography>
+                    <Typography variant="overline" color="text.secondary" component="div">
+                        {dateString}
+                    </Typography>
+                    <Typography variant="overline" color="text.secondary" component="div">
+                        {props.party.location}
+                    </Typography>
```
Comment (Satttoshi) [inline frontend/src/components/PartyItem.tsx:26]: Nice Mui usage 👍

**Review by Satttoshi — APPROVED** [review]
Some small suggestions, else very nice understandable code LGTM 🚀

File: backend/src/main/java/org/partypets/backend/service/PartyService.java (hunk 1 above)
Comment (toshydev) [reply backend/src/main/java/org/partypets/backend/service/PartyService.java:18]: We implemented it for the sole purpose of being able to mock it in the service unit tests. Do you have a suggestion on how else we could've solved this problem?

File: frontend/src/components/AddForm.tsx (hunk 2 above)
Comment (toshydev) [reply frontend/src/components/AddForm.tsx:12]: Isn't the type already set when using a default value? The states are all shown as `: string` in IntelliJ

File: frontend/src/components/PartyItem.tsx (hunk 5 above)
Comment (toshydev) [reply frontend/src/components/PartyItem.tsx:26]: Thanx!🙏 It's much less complicated than we thought💪

**Comment by toshydev** [comment]
> Some small suggestions, else very nice understandable code LGTM 🚀 

Thanks for the review & welcome to our code🤩
Left a question concerning generic typing☝️

File: backend/src/main/java/org/partypets/backend/service/PartyService.java (hunk 1 above)
Comment (Satttoshi) [reply backend/src/main/java/org/partypets/backend/service/PartyService.java:18]: You could mock your method, which generates the id, in whatever class you're invoking it.

File: frontend/src/components/AddForm.tsx (hunk 2 above)
Comment (Satttoshi) [reply frontend/src/components/AddForm.tsx:12]: You're right, IntelliJ is smart here, but imagine people on different IDEs. I don't actually know what's the best practice here, but I noticed that the mui code from the docs is indeed using generics for its useStates.

Comment (toshydev) [reply frontend/src/components/AddForm.tsx:12]: Good to know! 💪

**Pull request merged**